<?xml version="1.0"?>
<launch>

<arg name="name" default="yumi" doc="The robot name. Ensure this is the same name you give to the arm in the urdf instance."/>
<!--arg name="port" default="49939"/-->
<!--arg name="ip" default="192.168.1.1"/--> <!--when talking to a virtual controller -->
<arg name="ip" default="192.168.125.1"/> <!--when talking to the real robot controller -->
<arg name="hardware_interface" default="VelocityJointInterface"/>

<param name="robot_description" command="$(find xacro)/xacro.py '$(find yumi_description)/urdf/yumi.urdf.xacro' prefix:=$(arg hardware_interface)" />

<!-- the real hardware interface /-->
<node required="true" name="yumi_hw" pkg="yumi_hw" type="yumi_hw_ifce_node" respawn="false" ns="/yumi" output="screen"> <!--launch-prefix="xterm -e gdb - -args"-->
    <!-- addresses /-->
    <param name="name" value="$(arg name)" />
    <!--param name="port" value="$(arg port)"/-->
    <param name="ip" value="$(arg ip)"/>
</node>
 
<node required="true" name="yumi_gripper" pkg="yumi_hw" type="yumi_gripper_node" respawn="false" ns="/yumi" output="screen"> <!--launch-prefix="xterm -e gdb - -args"-->
    <!-- addresses /-->
    <param name="ip" value="$(arg ip)"/>
</node>
  <!-- Show in Rviz   -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find yumi_description)/launch/yumi.rviz"/>


  <arg name="controllers" default="joint_state_controller hiqp_kinematics_controller"/>
  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />
  <arg     if="$(arg debug)" name="launch_prefix"
     value="gdb --ex run --args" />


  <!-- Load the controllers -->

  <!-- Load the YAML file -->
  <rosparam file="$(find hiqp_controllers)/config/rbt_yumi.yaml" command="load"/>

  <!-- Load preload configurations for YuMi and HiQP -->
  <rosparam file="$(find hiqp_controllers)/config/yumi_hiqp_preload.yaml" command="load"/>

  <!-- Startup a spawner node that loads the controllers -->
  <node name="controller_spawner" 
        pkg="controller_manager" 
        type="spawner" 
        respawn="false" 
        output="screen"
        ns="/yumi"
        args="$(arg controllers)"/>

  <!-- Spawn the robot_state_publisher -->
  <node name="robot_state_publisher" 
        pkg="robot_state_publisher" 
        type="robot_state_publisher" 
        respawn="false" 
        output="screen">
    <remap from="/joint_states" to="/yumi/joint_states" />
  </node>

</launch>
